src/ralph/backends/data/async_es.py

```diff
@@ -129,6 +130,12 @@ async def read(  # noqa: PLR0913
             raw_output (bool): Controls whether to yield dictionaries or bytes.
             ignore_errors (bool): No impact as encoding errors are not expected in
                 Elasticsearch results.
+            prefetch: The number of records to prefetch (queue) while yielding.
+                If `prefetch` is `None` or `0` it defaults to `1` - no records are
```

Comment:
Maybe `prefetch` could default to 0, meaning no API call is made until it is consumed. `prefetch=1` would mean we fill a Queue of length 1 (we make one call as soon as `read` is called). Same for `prefetch>1` WDYT ?

Reply:
Good point) The current implementation relies on the `Queue` to fill and wait until it has free space.
The problem is that with a `Queue` of size `1`, it is filled when the first record is retrieved and awaited until it has free space after we try to put a second record in it - thus, we consume two records in advance instead of one (therefore we use `Queue(prefetch - 1)` and don't handle the case with one prefetched record.
Alternatively, we could handle the "waiting" ourselves by checking the queue size before the next iteration and awaiting the next item to be consumed to proceed. However, this might add additional complexity( Would this be preferred?

Reply:
I personally prefer having `prefetch` that defaults to `1`, and not making any queue in this case.